scripts/build.md

```diff
@@ -0,0 +1,115 @@
+# DOSBox build script
+
+A helper-script that builds DOSBox with varying compilers, release types, and versions
+on MacOS, Linux, and Windows.
+
+A second script, **list-build-dependencies.sh** prints a list of package and library
+dependencies used to build and link DOSBox, customized for your hardware, operating system,
+and selected compiler and its version.  You can use this script's output to install
+those necessary packages.
+
+## Requirements
+
+- **Windows newer than XP**
+    - **NTFS-based C:**, because msys2 doesn't work on FAT filesystems
+	- **Chocolately**, to install msys2
+
+- **MacOS** 10.x
+	- **brew**, as the package manager
+
+- **Ubuntu** 16.04 or newer
+	- **apt**, as the package manager
+	- **sudo**, to permit installation of depedencies (optional) 
+
+- **Build dependencies (all operating systems)**
+    - Per those listed by the accompanying list-build-dependencies.sh script
+
+## Windows Installation and Usage
+
+1. Download and install Chocolatey: https://chocolatey.org/install
+1. Open a console and run Cholocatey's command line interface (CLI) to install msys2:  
+   `choco install msys2 git --no-progress`
+
+    ```
+    Chocolatey v0.10.15
+    Installing the following packages:
+    msys2 git
+    By installing you accept licenses for the packages.
+    
+   msys2 v20180531.0.0 [Approved]
+   msys2 package files install completed. Performing other installation steps.
+   Installing to: C:\tools\msys64
+   Extracting 64-bit C:\ProgramData\chocolatey\lib\msys2\tools\msys2-base-x86_64-20180531.tar.xz to C:\tools\msys64...
+   C:\tools\msys64
+   Extracting C:\tools\msys64\msys2-base-x86_64-20180531.tar to C:\tools\msys64...
+   C:\tools\msys64
+   Starting initialization via msys2_shell.cmd
+   PATH environment variable does not have C:\tools\msys64 in it. Adding...
+   ```
+
+1. Open a new console and clone the DOSBox staging repository:  
+   `git clone https://github.com/dreamer/dosbox-staging.git`
+
+1. (optional) If you're interested in a specific branch:
+   * enter the repo directory: `cd dosbox-staging`
+   * list the available branches: `git branch --all`
+   * switch to a specific branch: `git checkout NAME_OF_BRANCH`
```

Comment:
Overall this file is great, thanks for documenting this! The only comment from my side is about this section - I don't think it's necessary. We're not in a business of writing Git tutorials here an and inexperienced user might decide that `git checkout remotes/origin/real/branch-name` is a good idea ;)